course/serializers.py

```diff
@@ -0,0 +1,135 @@
+from course.models import (
+    FlowSession, FlowPageVisit, FlowPageData, FlowPageVisitGrade)
+
+
+class FlowSessionSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = FlowSession
+
+        fields = ("id",
+                  "username",
+                  "institutional_id",
+                  "active_git_commit_sha",
+                  "flow_id",
+                  "start_time",
+                  "completion_time",
+                  "last_activity",
+                  "page_count",
+                  "in_progress",
+                  "access_rules_tag",
+                  "expiration_mode",
+                  "points",
+                  "max_points",
+                  "result_comment",
+                  "points_percentage",
+                  )
+
+    username = serializers.CharField(
+        source="participation.user.username", read_only=True)
+
+    institutional_id = serializers.CharField(
+        source="participation.user.institutional_id", read_only=True)
+
+    last_activity = serializers.SerializerMethodField()
+
+    points_percentage = serializers.SerializerMethodField()
+
+    def get_last_activity(self, obj):
+        return obj.last_activity()
+
+    def get_points_percentage(self, obj):
+        return obj.points_percentage()
+
+
+class FlowPageDateSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = FlowPageData
+
+        fields = ("page_ordinal",
+                  "page_type",
+                  "group_id",
+                  "page_id",
+                  "data",
+                  "title",
+                  "bookmarked"
+                  )
+
+
+class FlowPageVisitSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = FlowPageVisit
+
+        fields = ("flow_session",
+                  "page_data",
+                  "visit_time",
+                  "remote_address",
+                  "user",
+                  "impersonated_by",
+                  "is_synthetic",
+                  "answer",
+                  "is_submitted_answer",
+                  )
+
+    user = serializers.CharField(source="visitor.username", read_only=True)
+    impersonated_by = serializers.CharField(
+        source="impersonated_by.user.username", read_only=True)
+
+
+class FlowPageVisitGradeSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = FlowPageVisitGrade
+
+        fields = (
+            "visit",
+            "grader",
+            "grade_time",
+            "graded_at_git_commit_sha",
+            "grade_data",
+            "max_points",
+            "correctness",
+            "feedback",
+            "percentage",
+        )
+
+    grader = serializers.CharField(
+        source="grader.username", read_only=True)
+
+    percentage = serializers.SerializerMethodField()
+
+    def get_percentage(self, obj):
+        if obj.correctness is not None:
+            return 100 * obj.correctness
+        else:
+            return None
```

Comment:
Do we want to make this a read-only property on the `FlowPageVisitGrade` class instead of the serializer?

Reply:
Thanks for your review. What's your opinion? @inducer